ingest/Maybe.py

```diff
@@ -0,0 +1,55 @@
+__author__ = 'Stephan'
+
+from itertools import chain
+
+class Maybe:
+    def __init__(self, v=None):
+        self.value = v
+
+    @staticmethod
+    def nothing():
+        return Maybe()
+
+    @staticmethod
+    def of(t):
```

Comment:
I wrote the class and may will look into adding comments later.  I do not see adding comments to this class to be a blocker for this pull request.